app/models/page.rb

```diff
@@ -7,7 +7,7 @@ class Page < ApplicationRecord
 
   validates :title, presence: true
   validates :description, presence: true
-  validates :slug, presence: true, format: /\A[0-9a-z\-_]*\z/
+  validates :slug, presence: true, uniqueness: true, format: /\A[0-9a-z\-_]*\z/
```

Comment:
@kayamak thanks for this! I also ran into this same bug recently [while working on the Pages API](https://github.com/forem/forem/pull/19009/files#diff-92779dbf72cdd79815877e327b5693b0448053945e302d567814c33a5e0a49d1R14). This fix seems better than the one I wrote over there, I can adjust my PR if this one merges first. Ultimately, I suspect we may need to figure out a better configuration for [`unique_cross_model_slug`](https://github.com/forem/forem/blob/main/app/models/page.rb#L13).